src/Extensions/ElementalAreasExtension.php

```diff
@@ -257,6 +257,13 @@ public function ensureElementalAreasExist($elementalAreaRelations)
                 $area->OwnerClassName = get_class($this->owner);
                 $area->write();
                 $this->owner->$areaID = $area->ID;
+            }else{
+                // ensure correct OwnerClassName
+                $area = ElementalArea::get_by_id($this->owner->$areaID);
+                if( $area->OwnerClassName !== get_class($this->owner) ){
+                    $area->OwnerClassName = get_class($this->owner);
+                    $area->write();
+                }
```

Comment:
I think this doesn't really make sense in the context of `ensureElementalAreasExist`. We can't rename the method (which i'd prefer) but we can make a new method. Would you mind doing that?